### PR TITLE
Updated WMTS implementation and added a test.

### PR DESCRIPTION
**Pull request by rsupnekar**

### CODE REVIEW

**Review by ee11cbb19052e40b07aac0ca060c23ee — APPROVED** [review]

File: mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java [hunk 1]
```diff
@@ -584,70 +594,45 @@ public void creationFailed(LayerFactory factory, Layer layer, Throwable ex) {
...
+                    @Override
+                    public void creationSucceeded(LayerFactory factory, Layer layer) {
+                        // try to insert layer before the image layer
+                        final int insertIdx = ww.getLayers().indexOfLayer(MapInstance.this.imageLayer);
+                        if (insertIdx != -1) {
+                            ww.getLayers().addLayer(insertIdx, layer);
+                            MapInstance.this.wmtsHash.put(wmts.getGeoId(), layer);
+                            // redraw should be automatic
+                        } else {
+                            throw new IllegalStateException("ERROR: unable to locate tactical graphic layer.");
```
Comment (Ish-rivera) [inline mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java:623]: Update error "ERROR: unable to locate image layer."

File: mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java [hunk 2]
```diff
@@ -263,6 +265,7 @@ protected void initMapEngine(Context context, AttributeSet attrs) {
         this.oMapViewController = new Emp3NavigationListener(this, ww);
         this.handler = new Handler(Looper.getMainLooper());
         this.wmsHash = new HashMap<>();
+        this.wmtsHash = new HashMap<>();
```
Comment (Ish-rivera) [inline mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java:268]: Can't we have one mapServiceHash and add WMS and WMTS, all map service layers on it.

File: mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java [hunk 3]
```diff
@@ -584,70 +594,45 @@ public void creationFailed(LayerFactory factory, Layer layer, Throwable ex) {
...
-        WmsLayerConfig wmsConfig;
+    private void addWMTSService(final IWMTS wmts) {
+        // Create a layer factory, World Wind's general component for creating layers
+        // from complex data sources.
+        LayerFactory layerFactory = new LayerFactory();
         String sLayers = "";
-        String sStyles = "";
-
-        for (String Str : wms.getLayers()) {
+        for (String layerName : wmts.getLayers()) {
```
Comment (Ish-rivera) [inline mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java:602]: Can WMTS have more than 1 layer?

File: mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java [hunk 4]
```diff
@@ -584,70 +594,45 @@ public void creationFailed(LayerFactory factory, Layer layer, Throwable ex) {
...
+                sLayers,  // WMS layer name
+                new LayerFactory.Callback() {
+                    @Override
+                    public void creationSucceeded(LayerFactory factory, Layer layer) {
+                        // try to insert layer before the image layer
+                        final int insertIdx = ww.getLayers().indexOfLayer(MapInstance.this.imageLayer);
+                        if (insertIdx != -1) {
+                            ww.getLayers().addLayer(insertIdx, layer);
+                            MapInstance.this.wmtsHash.put(wmts.getGeoId(), layer);
+                            // redraw should be automatic
```
Comment (Ish-rivera) [inline mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/MapInstance.java:621]: we need to add it to the miniMap if its open.

File: sdk/sdk-api/src/main/java/mil/emp3/api/WMTS.java [hunk 5]
```diff
@@ -97,6 +97,8 @@ public void setDimensions(List<String> dimensions) {
      * @param newLayers A list of string. The list is ignored if its null or empty.
      * @throws EMP_Exception
      */
+
+    @Override
     public void setLayers(List<String> newLayers) throws EMP_Exception {
```
Comment (Ish-rivera) [inline sdk/sdk-api/src/main/java/mil/emp3/api/WMTS.java:102]: Can WMTS have more than 1 layer?

**Comment by rsupnekar** [comment]
Yes, we can have multiple layers per service.  ArcGIS supports that.

**Comment by rsupnekar** [comment]
To test, build this branch and run test basic.  From the menu, choose add WMTS.

**Comment by mubinakhan** [comment]
Reintegrate

**Pull request merged**